Docs/ns-frame (#619)
centrica-engineering/nucleus-docs: src/content/docs/components/ns-frame.mdx

---
title: Frame
---

export const componentName = "ns-frame";

import Specification from '@components/specification.astro';
import Placement from '@components/placement.astro';
import CompDetails from '@components/cem-description.astro';
import ExampleViewer from '@components/ExampleViewer.astro';
import RelatedLinks from '@components/related-links.astro';
import { Tabs, TabItem } from '@astrojs/starlight/components';

## Overview

<CompDetails name={componentName} />

The `ns-frame` component comprises a header and buttons that are adjacent to an anonymous content area. When a button is clicked, it navigates through the content. This component should not be used in conjunction with the `ns-header`. It is particularly useful in scenarios such as when a user is logged into their account.

### Examples

<ExampleViewer type="standard" />

## Guidance

:::do

> 

Reviewer: Remove whitespace
```suggestion
```

- Keep labels short, ideally single words.
- Minimise the number of menu options.
- Strive for concise labels.
- Ensure labels are composed of words that resonate with customers.
- Use this as the primary navigation.
- Utilise the account menu in the header for customer account management items.
- Employ menu items as overarching categories for customer content.

> 

Reviewer: Remove whitespace
```suggestion
```

:::

:::dont

> 

Reviewer: Remove whitespace
```suggestion
```

#### Key

| Key | Field type | Guidelines |
| :--- | :--- | :--- |
| <span class="marker">A</span> | Header | Frames the top of the page with a logo and an interactive icon. This icon, along with its label, does not serve general navigation purposes. Instead, it activates a drop-down menu that presents items such as account information. There are also various Search options to choose from - refer to the [examples](/components/ns-frame#examples) provided above. |
| <span class="marker">B</span> | Buttons | Each button includes an icon and a label and has states for active, hover, and selected. All buttons are contained within the menu, which becomes scrollable when the number of buttons exceeds the visible space. There is also the flexibility to hide all items, display as few as two, or show many items in the menu. |
| <span class="marker">C</span> | Menu scroll chevrons | These are interactive elements that provide visual cues and a method to navigate through the menu of buttons. This allows buttons positioned outside the visible area to be scrolled into view. These chevrons will automatically disappear if the number of buttons fits within the visible area. |

> | <span class="marker">D</span> | Anonymous slot | Include headings, lists and paragraphs. Divide content into short paragraphs as it is easier to consume, and aids effective page scanning. Include emphasis with `<strong>`, inline links `<a>`, lists `<ul>` `<ol>`. Add a [caveat link](/components/ns-caveat) if required. This should be followed by the [nsx-footer](/components/nsx-footer) to indicate the end of the page. |

Reviewer: ```suggestion
| <span class="marker">D</span> | Anonymous slot | To contain [`<ns-panel>`](/components/ns-panel). Pay attention to how the content displays at all resolutions. [`<nsx-footer>`](/components/nsx-footer) should be the last component. |
```

> | <span class="marker">E</span> | Search | The search feature can be displayed and allows personalisation based on the account credentials. |

Reviewer: ```suggestion
| <span class="marker">E</span> | Search | When Search is shown, it will contain the value of the `name` attribute, allowing for personalisation. Refer to the [`show-search` attribute](##specification-attribute-show-search) on how to use this option. |
```
